activerecord/lib/active_record/relation.rb

```diff
@@ -161,31 +161,27 @@ def first_or_initialize(attributes = nil, &block) # :nodoc:
     # failed due to validation errors it won't be persisted, you get what
     # #create returns in such situation.
     #
-    # Please note <b>this method is not atomic</b>, it runs first a SELECT, and if
```

Comment:
I think this warning is probably worth keeping... if you don't have a uniqueness constraint defined, it will mostly seem to work but have the described race.

Reply:
Hum, I don't think it should be kept as is, but yes, rather than to remove it I should rewod it to say that if you don't have a unique constaint, you'll end up with two records.

Reply:
Ok, I reworded the documentation, let me know what you think.